torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -4320,6 +4320,47 @@ def test_init_rpc_twice(self):
 
         rpc.shutdown()
 
+    # Test init_rpc without world_size argument
+    @dist_init(setup_rpc=False)
+    def test_init_rpc_without_world_size(self):
+        rpc.init_rpc(
+            name=worker_name(self.rank),
+            backend=self.rpc_backend,
+            rank=self.rank,
+            rpc_backend_options=self.rpc_backend_options,
+        )
+
+        # TODO: shutdown is ungraceful, not yet fully implemented
+        rpc.shutdown(graceful=False)
```

Comment:
Will this introduce any flakiness?

Reply:
Initially I thought it wouldn't since there is no communication and this test is only testing initialization. But now I can think of a case where Rank 0 shutdowns before another rank initializations, e.g. Rank 1. This would cause Rank 1 to hang and eventually error out.

Reply:
I added a sync before shutdown to solve this. Will remove once graceful shutdown is implemented.